data-service/src/component-tests/java/uk/gov/gchq/palisade/component/data/web/DataControllerTest.java

```diff
@@ -0,0 +1,133 @@
+
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.CompletableFuture;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.asyncDispatch;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.request;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.gchq.palisade.component.data.common.CommonTestData.AUDITABLE_DATA_REQUEST;
+import static uk.gov.gchq.palisade.component.data.common.CommonTestData.AUDITABLE_DATA_REQUEST_WITH_ERROR;
+import static uk.gov.gchq.palisade.component.data.common.CommonTestData.AUDITABLE_DATA_RESPONSE;
+import static uk.gov.gchq.palisade.component.data.common.CommonTestData.DATA_REQUEST;
+
+/**
+ * Tests for the DataController web service endpoint.
+ */
+@WebMvcTest(controllers = {DataController.class})
+@ContextConfiguration(classes = {DataControllerTest.class, DataController.class})
+class DataControllerTest {
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    @Autowired
+    private DataController controller;
+    @MockBean
+    private AuditableDataService serviceMock;
+    @MockBean
+    private AuditMessageService auditMessageServiceMock;
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    void testContextLoads() {
+        assertThat(controller).isNotNull();
+        assertThat(mockMvc).isNotNull();
+    }
+
+    /**
+     * Tests that the endpoint is expecting a Json string for a DataRequestModel.
+     * A return will be a OutputStream.
+     *
+     * @throws Exception if the test fails to run
+     */
+    @Test
+    void testControllerReturnsAccepted() throws Exception {
+        when(serviceMock.authoriseRequest(any()))
+                .thenReturn(CompletableFuture.completedFuture(AUDITABLE_DATA_REQUEST));
+
+        when(serviceMock.read(any(), any()))
+                .thenReturn(CompletableFuture.completedFuture(AUDITABLE_DATA_RESPONSE));
+
+        when(auditMessageServiceMock.auditMessage(any()))
+                .thenReturn(CompletableFuture.completedFuture(null));
+
+        MvcResult result = mockMvc.perform(post("/api/read/chunked")
+                .contentType("application/json")
+                .characterEncoding(StandardCharsets.UTF_8.name())
+                .content(MAPPER.writeValueAsBytes(DATA_REQUEST)))
+                .andExpect(request().asyncStarted())
+                .andReturn();
+
+        ResultActions resultActions = mockMvc.perform(asyncDispatch(result))
```

Comment:
If you're not using the result, why assign it?
```suggestion
        mockMvc.perform(asyncDispatch(result))
```